aip/client-libraries/4236.md

```diff
@@ -13,21 +13,38 @@ include the value of `google.api.api_version` in the request to the API.
 ### Expected Generator and Client Library Behavior
 
 If a service is annotated with `google.api.api_version`, client library
-generators **must** include the version in requests sent to the API via
-the HTTP Header `X-Goog-Api-Version`.
-
-Requests which contain the API version, **must** include either an HTTP query
-parameter `$apiVersion` or HTTP header `X-Goog-Api-Version`, but a request
-**must not** contain both.
+generators **must** include either an HTTP query parameter `$apiVersion`
+or HTTP header `X-Goog-Api-Version`, but a request **must not** contain both.
 
 Generated documentation for a given service **may** include the value of
 `google.api.api_version`, if it exists in the source protos.
 
 ## Rationale
 
-The `google.api.api_version` annotation allows services to abide by the schema and
-service behavior at the time the API version was deployed. The format of the
-API version **must** be treated as opaque by clients. Generators and clients **must not**
-interpret the value of `google.api.api_version` beyond the uses mentioned above.
+### Necessity for Versioning
+
+Explicit API versioning using the `google.api.api_version` annotation is essential
+for maintaining compatibility between clients and services over time. As services
+evolve, their schemas and behaviors may change. By specifying the API version, a
+client communicates its expectations to the service. This allows the service to
+respond in a manner consistent with the client's understanding, preventing errors
+or unexpected results due to incompatible changes.
+
+### Importance of Opaque Treatment
+
+Clients must treat the value of `google.api.api_version` as opaque to ensure robust
+compatibility.  This means that the specific format or structure of the version string
+should not be parsed or interpreted for any purpose beyond identifying the intended API
+version. By relying on this opaque identifier, clients avoid making assumptions about the
```

Comment:
So these first two sentences would be really good in the guidance section e.g.

```
Clients **must** treat the value of `gogle.api.api_version as opaque. Specifically, the value **must not**
be interpreted or parsed in any way.
```

Then this subsection could start:

```
Treating the `google.api.api_version` value as opaque is important to ensuring robust
compatibility guarantees. By relying on....
```

Reply:
Fixed in a228cba6a2a5a923473d284078e1a6986d144ca5